src/helpers/Fees.ts

```diff
@@ -0,0 +1,159 @@
+        ])
+
+        const dcaPercentage = dcaInvestments.reduce(
+            (acc, investment) => acc + investment.percentage,
+            BigInt(0),
+        )
+        const vaultPercentage = vaultInvestments.reduce(
+            (acc, investment) => acc + investment.percentage,
+            BigInt(0),
+        )
+        const liquidityPercentage = liquidityInvestments.reduce(
+            (acc, investment) => acc + investment.percentage,
+            BigInt(0),
+        )
+        const tokensPercentage = tokenInvestments.reduce(
+            (acc, investment) => acc + investment.percentage,
+            BigInt(0),
+        )
+
+        const [
+            strategistPercentage,
+            dcaBaseFeeBP,
+            vaultBaseFeeBP,
+            liquidityBaseFeeBP,
+            exchangeBaseFeeBP,
+            dcaNonSubscriberFeeBP,
+            vaultNonSubscriberFeeBP,
+            liquidityNonSubscriberFeeBP,
+            exchangeNonSubscriberFeeBP,
+        ] = await Promise.all([
+            isHottestDeal
+                ? strategyManager.hotStrategistPercentage()
+                : strategyManager.strategistPercentage(),
+            dca.baseFeeBP(),
+            vaultManager.baseFeeBP(),
+            liquidityManager.baseFeeBP(),
+            exchangeManager.baseFeeBP(),
+            subscribedUser ? BigInt(0) : dca.nonSubscriberFeeBP(),
+            subscribedUser ? BigInt(0) : vaultManager.nonSubscriberFeeBP(),
+            subscribedUser ? BigInt(0) : liquidityManager.nonSubscriberFeeBP(),
+            subscribedUser ? BigInt(0) : exchangeManager.nonSubscriberFeeBP(),
+        ])
+
+        const baseFee = new BigNumber(
+            (
+                dcaBaseFeeBP * dcaPercentage +
+                vaultBaseFeeBP * vaultPercentage +
+                liquidityBaseFeeBP * liquidityPercentage +
+                exchangeBaseFeeBP * tokensPercentage
+            ).toString(),
+        ).div(10_000)
+        const nonSubscriberFee = new BigNumber(
+            (
+                dcaNonSubscriberFeeBP * dcaPercentage +
+                vaultNonSubscriberFeeBP * vaultPercentage +
+                liquidityNonSubscriberFeeBP * liquidityPercentage +
+                exchangeNonSubscriberFeeBP * tokensPercentage
+            ).toString(),
+        ).div(10_000)
+        const strategistFee = baseFee.times(strategistPercentage.toString()).div(100)
+
+        return {
+            protocolFee: baseFee.minus(strategistFee).plus(nonSubscriberFee),
+            strategistFee,
+        }
+    }
+
+    public static async getStrategyFeeAmount(
+        amount: bigint,
+        strategyManager: StrategyManager,
+        strategyId: BigNumberish,
+        subscribedUser: boolean,
+        dca: UseFee,
+        vaultManager: UseFee,
+        liquidityManager: UseFee,
+        exchangeManager: UseFee,
+    ) {
+        const {
+            protocolFee,
+            strategistFee,
+        } = await this.getStrategyFeePercentage(
```

Comment:
Avoid unexpected issues with:
```suggestion
        } = await Fees.getStrategyFeePercentage(
```